pipeline/trusted-resources/pkg/trustedtask/trustedtask.go

```diff
@@ -0,0 +1,82 @@
+package trustedtask
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	"github.com/tektoncd/pipeline/pkg/client/injection/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/apis"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
+)
+
+type TrustedTask struct {
+	v1beta1.Task
+}
+
+func (t *TrustedTask) Validate(ctx context.Context) (errs *apis.FieldError) {
+	k8sClient := kubeclient.Get(ctx)
+	tektonClient := client.Get(ctx)
+
+	if err := t.verifyTask(ctx, k8sClient, tektonClient); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Non blocking for this PR, but this is pretty trivial now - it may make sense to collapse this down to 1 func.

Reply:
thanks Billy. so here mean we could arrange `verifyTask` into `Validate`, so having a single func right?